homeassistant/components/airly/sensor.py

```diff
@@ -0,0 +1,154 @@
+"""Support for the Airly sensor service."""
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    CONF_NAME,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    PRESSURE_HPA,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_API_HUMIDITY,
+    ATTR_API_PM1,
+    ATTR_API_PRESSURE,
+    ATTR_API_TEMPERATURE,
+    DATA_CLIENT,
+    DOMAIN,
+)
+
+ATTRIBUTION = "Data provided by Airly"
+
+ATTR_ICON = "icon"
+ATTR_LABEL = "label"
+ATTR_UNIT = "unit"
+
+HUMI_PERCENT = "%"
+VOLUME_MICROGRAMS_PER_CUBIC_METER = "µg/m³"
+
+SENSOR_TYPES = {
+    ATTR_API_PM1: {
+        ATTR_DEVICE_CLASS: None,
+        ATTR_ICON: "mdi:blur",
+        ATTR_LABEL: ATTR_API_PM1,
+        ATTR_UNIT: VOLUME_MICROGRAMS_PER_CUBIC_METER,
+    },
+    ATTR_API_HUMIDITY: {
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
+        ATTR_ICON: "mdi:water-percent",
+        ATTR_LABEL: ATTR_API_HUMIDITY.capitalize(),
+        ATTR_UNIT: HUMI_PERCENT,
+    },
+    ATTR_API_PRESSURE: {
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_PRESSURE,
+        ATTR_ICON: "mdi:gauge",
+        ATTR_LABEL: ATTR_API_PRESSURE.capitalize(),
+        ATTR_UNIT: PRESSURE_HPA,
+    },
+    ATTR_API_TEMPERATURE: {
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+        ATTR_ICON: "mdi:thermometer",
```

Comment:
We should not add icons for things that have device classes. The frontend should do this automatically. 

Referencing a single `mdi:` icon causes the frontend to load a 1MB icon file 😞

Reply:
Didn't know that. Change done.